pkg/catalog/catalog.go

```diff
@@ -974,6 +979,24 @@ func (c *Catalog) GetCommit(ctx context.Context, repositoryID string, reference
 	return catalogCommitLog, nil
 }
 
+// wrapper for simplelru.LRU that is safe to access from multiple go routines
+type safeCache struct {
```

Comment:
Can be removed and use 'github.com/hnlq715/golang-lru' - the current lru is the underlaying storage.